llvm/lib/Transforms/InstCombine/InstCombineMulDivRem.cpp

```diff
@@ -1748,6 +1748,18 @@ Instruction *InstCombinerImpl::visitFDiv(BinaryOperator &I) {
 
   if (Instruction *Mul = foldFDivPowDivisor(I, Builder))
     return Mul;
+  Value *ExpX;
+  // exp(exp(X)) / exp(X) -> exp(exp(X) - X)
```

Comment:
Not seeing any checks for FMF here? Please add negative tests that the transform is not performed without needed flags.